index.js

```diff
@@ -72,6 +74,25 @@ function dispatcher() {
     invalid(`command doesn't exist: ${command}`);
   }
 
+  const username = configFile.get("email");
+  const isOptedIn = configFile.get(OPT_IN_NAME) || false;
+
+  // track only if email is available and user is opted In
+  if (isOptedIn && username) {
+    init(AMPLITUDE_API_KEY);
+
+    // define the properties to track
+    const eventProperties = {
+      full_command: process.argv.slice(2).join(" "), // all of the commands in the user input
+      action: command // Just the first command
+    };
+
+    // track the event
+    track("Crowdbotics CLI", eventProperties, {
+      user_id: username
+    });
+  }
```

Comment:
Could this maybe all be in a separate function? 

And if possible, all inside a try/catch. Maybe do nothing if it error out for now...  If something is messed up with Amplitude, we don't want the user to be blocked in their actions. Later on we can track errors somehow.

Reply:
I put an extra property to capture the error through another tracking event... though it is questionable if the try fails then can the catch actually send an event to amplitude properly.

Probably need to rethink this.

Reply:
It might not be a bad idea, maybe if we group it as a different event like `Crowdbotics CLI Errors`. But we could check with product too